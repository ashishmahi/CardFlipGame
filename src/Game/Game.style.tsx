import styled from 'styled-components/native';

export const RestartText = styled.Text`
  color: blue;
  font-size: 20px;
`;

export const Steps = styled.Text`
  font-size: 30px;
  color: white;
`;

export const StepCount = styled.Text`
  font-size: 40px;
  color: deepskyblue;
`;

export const RestartAndStepsContainer = styled.View`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 16px;
`;

export const GameGrid = styled.View`
  flex-direction: row;
  flex-wrap: wrap;
`;
