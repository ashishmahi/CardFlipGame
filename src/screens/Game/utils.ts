import {CardModal} from '../../components/Card/Card';

const generateRandomCards: () => CardModal[] = () => {
  const cards: CardModal[] = [];
  while (cards.length < 12) {
    var number = Math.floor(Math.random() * 100);
    if (cards.findIndex(card => card.number === number) === -1) {
      cards.push({number, id: cards.length});
      cards.push({number, id: cards.length + 6});
    }
  }
  return cards;
};

function shuffle<T>(array: T[]): T[] {
  for (let i = array.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [array[i], array[j]] = [array[j], array[i]];
  }
  return array;
}

// generate 6 random pairs (6 uniques and duplicate it)
// shuffle it
export const generateGrid: () => CardModal[] = () => {
  const randomPairs = generateRandomCards();
  return shuffle(randomPairs);
};
