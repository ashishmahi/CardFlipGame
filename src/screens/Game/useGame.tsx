import React from 'react';
import {Alert} from 'react-native';
import {CardModal} from '../../components/Card/Card';
import {generateGrid} from './utils';

export const useGame = () => {
  // const gridNumbers: CardModal[] = [
  //   {number: 1, id: 0},
  //   {number: 2, id: 1},
  //   {number: 3, id: 2},
  //   {number: 4, id: 3},
  //   {number: 5, id: 4},
  //   {number: 6, id: 5},
  //   {number: 1, id: 6},
  //   {number: 2, id: 7},
  //   {number: 3, id: 8},
  //   {number: 4, id: 9},
  //   {number: 5, id: 10},
  //   {number: 6, id: 11},
  // ];
  const [gridNumbers] = React.useState<CardModal[]>(generateGrid());
  const [currentOpenPair, setCurrentOpenPair] = React.useState<CardModal[]>([]);
  const [noOfSteps, setNoOfSteps] = React.useState<number>(0);
  const [matchedCards, setMatchedCards] = React.useState<CardModal[]>([]);

  const onCardClick = (card: CardModal) => {
    const possibleMatchingPair = currentOpenPair.concat([card]);
    setCurrentOpenPair(possibleMatchingPair);
    const isAnyCardOpenForMatching = currentOpenPair.length;
    if (isAnyCardOpenForMatching) {
      const matchFailed = card.number !== currentOpenPair[0].number;
      if (matchFailed) {
        setTimeout(() => {
          setCurrentOpenPair([]);
        }, 1000);
      } else {
        const updatedMatchedCard = matchedCards.concat(possibleMatchingPair);
        setMatchedCards(updatedMatchedCard);
        setCurrentOpenPair([]);
        const hasGameWon = updatedMatchedCard.length === 12;
        if (hasGameWon) {
          Alert.alert(
            'Congratulation',
            `You win this game in ${noOfSteps + 1} steps`,
            [
              {
                text: 'Try Another Round',
                onPress: () => {
                  console.log('calling restart');
                  restartGame();
                },
              },
            ],
          );
        }
      }
    }
    setNoOfSteps(noOfSteps + 1);
  };

  const restartGame = () => {
    setCurrentOpenPair([]);
    setNoOfSteps(0);
    setMatchedCards([]);
  };

  const isCardOpen = (cardId: number) => {
    const isCurrentCardOpen = Boolean(
      currentOpenPair.find(card => card.id === cardId),
    );

    const isCardResolved = Boolean(
      matchedCards.find(card => card.id === cardId),
    );
    return isCurrentCardOpen || isCardResolved;
  };

  return {
    gridNumbers,
    onCardClick,
    isCardOpen,
    noOfSteps,
    restartGame,
  };
};

export default useGame;
