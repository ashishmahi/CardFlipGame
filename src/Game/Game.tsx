import React from 'react';
import {Pressable} from 'react-native';
import Card from './components/Card/Card';
import {
  GameGrid,
  RestartAndStepsContainer,
  RestartText,
  StepCount,
  Steps,
} from './Game.style';
import useGame from './useGame';

const Game: React.FC = () => {
  const {gridNumbers, onCardClick, isCardOpen, noOfSteps} = useGame();
  return (
    <>
      <RestartAndStepsContainer>
        <Pressable>
          <RestartText>Restart</RestartText>
        </Pressable>
        <Steps>
          STEPS:<StepCount testID="stepCounter">{noOfSteps}</StepCount>{' '}
        </Steps>
      </RestartAndStepsContainer>
      <GameGrid>
        {gridNumbers.map(card => {
          return (
            <Card
              onCardPress={() => onCardClick(card)}
              key={`id is ${card.id}`}
              card={card}
              isOpen={isCardOpen(card.id)}
            />
          );
        })}
      </GameGrid>
    </>
  );
};

export default Game;
