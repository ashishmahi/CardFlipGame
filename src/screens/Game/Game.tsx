import React from 'react';
import {Pressable} from 'react-native';
import Card from '../../components/Card/Card';
import {
  GameGrid,
  RestartAndStepsContainer,
  RestartText,
  StepCount,
  Steps,
} from './Game.style';
import useGame from './useGame';

const Game: React.FC = () => {
  const {grid, onCardClick, isCardOpen, noOfSteps, restartGame} = useGame();
  return (
    <>
      <RestartAndStepsContainer>
        <Pressable testID="restart" onPress={restartGame}>
          <RestartText>Restart</RestartText>
        </Pressable>
        <Steps>
          STEPS:<StepCount testID="stepCounter">{noOfSteps}</StepCount>{' '}
        </Steps>
      </RestartAndStepsContainer>
      <GameGrid>
        {grid.map(card => {
          return (
            <Card
              onCardPress={() => onCardClick(card)}
              key={`id-${card.id}`}
              card={card}
              isOpen={isCardOpen(card.id)}
            />
          );
        })}
      </GameGrid>
    </>
  );
};

export default Game;
