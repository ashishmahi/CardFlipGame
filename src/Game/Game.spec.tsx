import React from 'react';
import {fireEvent, render, RenderResult} from '@testing-library/react-native';
import Game from './Game';
import {Alert} from 'react-native';

jest.useFakeTimers();

describe('Game Test', () => {
  const winGame = (screen: RenderResult) => {
    const cell0 = screen.getByTestId('card-0');
    const cell1 = screen.getByTestId('card-1');
    const cell2 = screen.getByTestId('card-2');
    const cell3 = screen.getByTestId('card-3');
    const cell4 = screen.getByTestId('card-4');
    const cell5 = screen.getByTestId('card-5');
    const cell6 = screen.getByTestId('card-6');
    const cell7 = screen.getByTestId('card-7');
    const cell8 = screen.getByTestId('card-8');
    const cell9 = screen.getByTestId('card-9');
    const cell10 = screen.getByTestId('card-10');
    const cell11 = screen.getByTestId('card-11');

    fireEvent.press(cell0);
    fireEvent.press(cell6);

    fireEvent.press(cell1);
    fireEvent.press(cell7);

    fireEvent.press(cell2);
    fireEvent.press(cell8);

    fireEvent.press(cell3);
    fireEvent.press(cell9);

    fireEvent.press(cell4);
    fireEvent.press(cell10);

    fireEvent.press(cell5);
    fireEvent.press(cell11);

    jest.runAllTimers();
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render  Game', () => {
    expect(render(<Game />)).toMatchSnapshot();
  });
  it('should open the card when clicked', () => {
    const container = render(<Game />);
    const cell = container.getByTestId('card-1');
    const cellContent = container.getByTestId('card-content-1');
    fireEvent.press(cell);
    expect(cell.props.style[0].backgroundColor).toBe('white');
    expect(cellContent.children).toContain('2');
  });

  it('should open only single card', () => {
    const container = render(<Game />);
    const cell = container.getByTestId('card-2');
    fireEvent.press(cell);
    expect(container.getAllByText('3')).toHaveLength(1);
  });

  it('should increase steps counter', () => {
    const container = render(<Game />);
    const cell = container.getByTestId('card-1');
    fireEvent.press(cell);
    expect(container.getByTestId('stepCounter').children).toContain('1');
  });

  it('should open, wait for a second and close both the card if pair is non matching', () => {
    const container = render(<Game />);
    const cell1 = container.getByTestId('card-1');
    const cell2 = container.getByTestId('card-2');
    fireEvent.press(cell1);
    fireEvent.press(cell2);
    expect(cell1.props.style[0].backgroundColor).toBe('white');
    expect(cell2.props.style[0].backgroundColor).toBe('white');

    jest.runAllTimers();
    expect(cell1.props.style[0].backgroundColor).toBe('deepskyblue');
    expect(cell2.props.style[0].backgroundColor).toBe('deepskyblue');
  });

  it('should keep open the matching pairs', () => {
    const container = render(<Game />);
    const cell1 = container.getByTestId('card-1');
    const cell2 = container.getByTestId('card-7');
    fireEvent.press(cell1);
    fireEvent.press(cell2);
    expect(cell1.props.style[0].backgroundColor).toBe('white');
    expect(cell2.props.style[0].backgroundColor).toBe('white');

    jest.runAllTimers();
    expect(cell1.props.style[0].backgroundColor).toBe('white');
    expect(cell2.props.style[0].backgroundColor).toBe('white');
  });

  it('should keep open the matching pairs even after clicking on third card', () => {
    const container = render(<Game />);
    const cell1 = container.getByTestId('card-1');
    const cell7 = container.getByTestId('card-7');
    const cell2 = container.getByTestId('card-2');
    fireEvent.press(cell1);
    fireEvent.press(cell7);
    expect(cell1.props.style[0].backgroundColor).toBe('white');
    expect(cell7.props.style[0].backgroundColor).toBe('white');

    jest.runAllTimers();
    fireEvent.press(cell2);
    jest.runAllTimers();
    expect(cell1.props.style[0].backgroundColor).toBe('white');
    expect(cell7.props.style[0].backgroundColor).toBe('white');
  });

  it('should alert on winning the game', () => {
    const spyAlert = jest.spyOn(Alert, 'alert');
    const container = render(<Game />);
    winGame(container);
    expect(spyAlert).toHaveBeenCalledWith(
      'Congratulation',
      'You win this game in 12 steps',
      [{onPress: expect.any(Function), text: 'Try Another Round'}],
    );
  });
  it('should restart game on restart click', () => {
    const container = render(<Game />);
    const cell1 = container.getByTestId('card-1');
    const cell7 = container.getByTestId('card-7');

    fireEvent.press(cell1);
    fireEvent.press(cell7);
    fireEvent.press(container.getByTestId('restart'));

    expect(cell1.props.style[0].backgroundColor).toBe('deepskyblue');
    expect(cell7.props.style[0].backgroundColor).toBe('deepskyblue');
    expect(container.getByTestId('stepCounter').children).toContain('0');
  });
  it('should restart game on restart click of winning message', () => {
    const spyAlert = jest.spyOn(Alert, 'alert');
    const container = render(<Game />);

    winGame(container);

    spyAlert.mock.calls[0][2][0].onPress();

    const cell1 = container.getByTestId('card-1');
    const cell7 = container.getByTestId('card-7');

    expect(cell1.props.style[0].backgroundColor).toBe('deepskyblue');
    expect(cell7.props.style[0].backgroundColor).toBe('deepskyblue');

    expect(container.getByTestId('stepCounter').children).toContain('0');
  });
});
