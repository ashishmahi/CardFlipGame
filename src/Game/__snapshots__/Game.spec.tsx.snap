// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Test should render  Game 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          Array [
            Object {
              "color": "blue",
              "fontSize": 20,
            },
          ]
        }
      >
        Restart
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 30,
          },
        ]
      }
    >
      STEPS:
      <Text
        style={
          Array [
            Object {
              "color": "deepskyblue",
              "fontSize": 40,
            },
          ]
        }
        testID="stepCounter"
      >
        0
      </Text>
       
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      isOpen={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "deepskyblue",
            "borderColor": "white",
            "borderStyle": "solid",
            "borderWidth": 4,
            "height": 140,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": "27%",
          },
        ]
      }
      testID="card-0"
    >
      <Text
        isOpen={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
        testID="card-content-0"
      >
        ?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      isOpen={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "deepskyblue",
            "borderColor": "white",
            "borderStyle": "solid",
            "borderWidth": 4,
            "height": 140,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": "27%",
          },
        ]
      }
      testID="card-1"
    >
      <Text
        isOpen={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
        testID="card-content-1"
      >
        ?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      isOpen={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "deepskyblue",
            "borderColor": "white",
            "borderStyle": "solid",
            "borderWidth": 4,
            "height": 140,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": "27%",
          },
        ]
      }
      testID="card-2"
    >
      <Text
        isOpen={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
        testID="card-content-2"
      >
        ?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      isOpen={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "deepskyblue",
            "borderColor": "white",
            "borderStyle": "solid",
            "borderWidth": 4,
            "height": 140,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": "27%",
          },
        ]
      }
      testID="card-3"
    >
      <Text
        isOpen={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
        testID="card-content-3"
      >
        ?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      isOpen={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "deepskyblue",
            "borderColor": "white",
            "borderStyle": "solid",
            "borderWidth": 4,
            "height": 140,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": "27%",
          },
        ]
      }
      testID="card-4"
    >
      <Text
        isOpen={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
        testID="card-content-4"
      >
        ?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      isOpen={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "deepskyblue",
            "borderColor": "white",
            "borderStyle": "solid",
            "borderWidth": 4,
            "height": 140,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": "27%",
          },
        ]
      }
      testID="card-5"
    >
      <Text
        isOpen={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
        testID="card-content-5"
      >
        ?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      isOpen={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "deepskyblue",
            "borderColor": "white",
            "borderStyle": "solid",
            "borderWidth": 4,
            "height": 140,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": "27%",
          },
        ]
      }
      testID="card-6"
    >
      <Text
        isOpen={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
        testID="card-content-6"
      >
        ?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      isOpen={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "deepskyblue",
            "borderColor": "white",
            "borderStyle": "solid",
            "borderWidth": 4,
            "height": 140,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": "27%",
          },
        ]
      }
      testID="card-7"
    >
      <Text
        isOpen={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
        testID="card-content-7"
      >
        ?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      isOpen={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "deepskyblue",
            "borderColor": "white",
            "borderStyle": "solid",
            "borderWidth": 4,
            "height": 140,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": "27%",
          },
        ]
      }
      testID="card-8"
    >
      <Text
        isOpen={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
        testID="card-content-8"
      >
        ?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      isOpen={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "deepskyblue",
            "borderColor": "white",
            "borderStyle": "solid",
            "borderWidth": 4,
            "height": 140,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": "27%",
          },
        ]
      }
      testID="card-9"
    >
      <Text
        isOpen={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
        testID="card-content-9"
      >
        ?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      isOpen={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "deepskyblue",
            "borderColor": "white",
            "borderStyle": "solid",
            "borderWidth": 4,
            "height": 140,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": "27%",
          },
        ]
      }
      testID="card-10"
    >
      <Text
        isOpen={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
        testID="card-content-10"
      >
        ?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      isOpen={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "deepskyblue",
            "borderColor": "white",
            "borderStyle": "solid",
            "borderWidth": 4,
            "height": 140,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "width": "27%",
          },
        ]
      }
      testID="card-11"
    >
      <Text
        isOpen={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
        testID="card-content-11"
      >
        ?
      </Text>
    </View>
  </View>,
]
`;
